compiler/rustc_resolve/src/late.rs

```diff
@@ -2768,7 +2777,25 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
         };
 
         let res = binding.res();
-        let Res::Def(def_kind, _) = res else { bug!() };
+        let Res::Def(def_kind, id_in_trait) = res else { bug!() };
+
+        match seen_trait_items[ns].entry(id_in_trait) {
```

Comment:
```suggestion
        match seen_trait_items.entry(id_in_trait) {
```
I don't think `PerNS` is necessary here?
`DefId`s of the trait items should be disjoint anyway.